src/types/tuple.rs

```diff
@@ -81,10 +84,44 @@ impl Tuple {
             } else {
                 let mut value_bytes = value.to_raw();
 
-                if value.is_variable() {
-                    bytes.append(&mut (value_bytes.len() as u32).encode_fixed_vec());
+                let col = &self.columns[i];
+                match col.desc.column_datatype {
+                    LogicalType::Boolean
+                    | LogicalType::Tinyint
+                    | LogicalType::UTinyint
+                    | LogicalType::Smallint
+                    | LogicalType::USmallint
+                    | LogicalType::Integer
+                    | LogicalType::UInteger
+                    | LogicalType::Bigint
+                    | LogicalType::UBigint
+                    | LogicalType::Decimal(_, _)
+                    | LogicalType::Float
+                    | LogicalType::Double
+                    | LogicalType::Date
+                    | LogicalType::DateTime => bytes.append(&mut value_bytes),
+                    LogicalType::Varchar(_) => {
+                        bytes.append(&mut (value_bytes.len() as u32).encode_fixed_vec());
+                        bytes.append(&mut value_bytes);
+                    }
+                    LogicalType::Text => {
+                        // len of text file
+                        let len = value_bytes.len();
```

Comment:
This length is not the overall length of the data, but the length stored in the Tuple. Because Text indirectly points to the Text file, the len here should not be value_bytes.len(), but the length of prefix+file_path (`deserialize_from` required)